.github/workflows/droid.yml

```diff
@@ -0,0 +1,64 @@
+name: Build for Android
+
+on:
+  workflow_dispatch:
+    inputs:
+      release_type:
+        required: true
+        type: string
+  workflow_call:
+    inputs:
+      release_type:
+        required: true
+        type: string
+    secrets:
+      GCP_CREDENTIALS:
+        required: true
+
+env:
+  NDK_ZIP: android-ndk-r26-linux.zip
+  TARGET: bin/viam-server-android-${{ inputs.release_type }}
+
+jobs:
+  droid:
+    runs-on: ubuntu-latest
```

Comment:
Is there a reason to not use canon (and a buildjet runner) for this? I see you're installing go, and if I'm not mistaken, this is doing 90% of the same build that the normal server build does, so it would also make full use of all the cached go packages, as well as having go already installed too.

(Eventually, a docker image for android builds might be a good idea, or adding NDK to the main viam-rdk image you're working on, depending on how much it bloats things.)

Reply:
`actions/setup-go` does its own caching -- fully cached, the build takes 1:30 on a standard github runner

not using canon because this has no CGo library dependencies

Reply:
I meant the go packages themselves, not the CGO stuff. There's several hundred megabytes of downloads in ~/go/pkg/ needed to build viam-server, but I didn't realize that setup-go now has full caching for all that (and not just the golang install itself.) Cool!